### PR TITLE
Add BTC price fetching

### PR DESCRIPTION
**Pull request by sreefoundry**
feat:
- add BTC price fetch method from yfinance
- add WBTC price fetch method from DEX Screener

### CODE REVIEW

**Review by rahulfoundry — DISMISSED** [review]
LGTM

**Review by rahulfoundry — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,104 @@
-bittensor>=7
-starlette>=0.30.0
-pydantic>=2
-rich>=13
-pytest>=8
-torch>=2
-numpy>=1
-setuptools>=68
\ No newline at end of file
+aiohappyeyeballs==2.3.4
```
Comment (mtrudeau-foundry-digital) [inline requirements.txt:1]: it looks like you only added yahoo finance as a dependency. please only add yahoo finance to the requirements text file

File: template/utils/btc_price.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Args:
+        str (period): Time period for price data (always "1d" for daily predictions).
+        str (interval): Time interval for price data ("5m", "30m", "1h", "4h", or "24h").
+
+    Returns:
+        float: BTC price in USD.
+    """
+    btc = yf.Ticker("BTC-USD")
+    btc_hist = btc.history(period=period, interval=interval)
+    print(btc_hist)
```
Comment (mtrudeau-foundry-digital) [inline template/utils/btc_price.py:32]: do we need this print statement?

File: template/utils/btc_price.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+def btc_price(period: str, interval: str) -> DataFrame:
+    """Fetches Bitcoin (BTC) price from yfinance.
+
+    Args:
+        str (period): Time period for price data (always "1d" for daily predictions).
+        str (interval): Time interval for price data ("5m", "30m", "1h", "4h", or "24h").
+
+    Returns:
+        float: BTC price in USD.
```
Comment (mtrudeau-foundry-digital) [inline template/utils/btc_price.py:28]: are we returning a float or a dataframe?

**Review by mtrudeau-foundry-digital — CHANGES_REQUESTED** [review]

**Review by mtrudeau-foundry-digital — APPROVED** [review]

**Pull request merged**